validation/src/main/scala/hmda/validation/dsl/PredicateGeo.scala

```diff
@@ -0,0 +1,114 @@
+package hmda.validation.dsl
+
+import hmda.model.census.{ CBSAMetroMicroLookup, CBSATractLookup }
+import hmda.model.fi.lar.Geography
+
+object PredicateGeo {
+  val cbsaTracts = CBSATractLookup.values
+
+  val cbsaMetroMicro = CBSAMetroMicroLookup.values
+
+  val validMsaCombinationSet = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }.toSet
+
+  val validMdCombinationSet = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }.toSet
+
+  val validMsaCombinationSetNoTract = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county)
+  }.toSet
+
+  val validMdCombinationSetNoTract = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county)
+  }.toSet
+
+  val hasMsaSet = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county)
+  }.filter(x => x._1 != "").map { x =>
+    (x._2, x._3)
+  }.toSet
+
+  val hasMdSet = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county)
+  }.filter(x => x._1 != "").map { x =>
+    (x._2, x._3)
+  }.toSet
+
+  val MsaNotMicro = cbsaMetroMicro.map { cbsa =>
+    (cbsa.GEOIOD, cbsa.MEMI)
+  }.filter(x => x._2 == 1).map { x =>
+    x._1
+  }
+
+  val hasMsaNotMicroSet = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county)
+  }.filter(x => MsaNotMicro.contains(x._1)).map { x =>
+    (x._2, x._3)
+  }.toSet
+
+  val validStateCountyTractCombinationSet = cbsaTracts.map { cbsa =>
+    (cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }.toSet
+
+  val validStateCountyCombinationSet = cbsaTracts.map { cbsa =>
+    (cbsa.state, cbsa.county)
+  }.toSet
+
+  val smallCounties = cbsaTracts
+    .filter { cbsa => cbsa.smallCounty == 1 }
+    .map { cbsa => (cbsa.state, cbsa.county) }
+    .toSet
+
+  implicit def smallCounty: Predicate[Geography] = new Predicate[Geography] {
+    override def validate: (Geography) => Boolean = _.asInstanceOf[AnyRef] match {
+      case geo: Geography => smallCounties.contains((geo.state, geo.county))
+      case _ => false
+    }
```

Comment:
I prefer something like this (or perhaps even less code could work, but this certainly does):
```scala
    override def validate: (Geography) => Boolean = (geo) => smallCounties.contains((geo.state, geo.county))
```
Same deal for all the following predicates.
This may also improve the code coverage numbers. ;-)